stdlib/src/utils/string_slice.mojo

```diff
@@ -1317,15 +1317,13 @@ fn _split_impl[
         iterator = src_str.__iter__()
         i_len = len(iterator) + 2
         out_ptr = UnsafePointer[Span[Byte, O]].alloc(i_len)
-        out_ptr.init_pointee_move(rebind[Span[Byte, O]]("".as_bytes_read()))
+        out_ptr.init_pointee_move(src_str.as_bytes_read()[:0])
         i = 1
         for s in iterator:
             sp = rebind[Span[Byte, O]](s.as_bytes_read())
             (out_ptr + i).init_pointee_move(sp^)
             i += 1
-        (out_ptr + i).init_pointee_move(
-            rebind[Span[Byte, O]]("".as_bytes_read())
-        )
+        (out_ptr + i).init_pointee_move(src_str.as_bytes_read()[:0])
```

Comment:
This should technically point to the end of the span (something like `[len(..) - 1:]`)?

And since `Span` is trivial, you can simply write `out_ptr[i] = rebind[Span[Byte, O]](s.as_bytes_read())`.

Reply:
thanks for the sugestions, went for this:
```mojo
out_ptr[0] = src_str.as_bytes_read()[0:0]
i = 1
for s in iterator:
    out_ptr[i] = rebind[Span[Byte, O]](s.as_bytes_read())
    i += 1
out_ptr[i] = src_str.as_bytes_read()[-1:-1]
```